src/channel_internal.c

```diff
@@ -78,27 +78,36 @@ static void abandon_pending_operations(void* context)
 {
     CHANNEL_INTERNAL* channel_internal_ptr = context;
 
+    DLIST_ENTRY op_list;
+    DList_InitializeListHead(&op_list);
     /*Codes_SRS_CHANNEL_INTERNAL_43_167: [abandon_pending_operations shall call srw_lock_acquire_exclusive.]*/
     srw_lock_acquire_exclusive(channel_internal_ptr->lock);
     {
         /*Codes_SRS_CHANNEL_INTERNAL_43_168: [abandon_pending_operations shall set is_open to false.]*/
         channel_internal_ptr->is_open = false;
 
+        /*Codes_SRS_CHANNEL_INTERNAL_43_174: [abandon_pending_operations shall make a local copy of the list of pending operations.]*/
+        DList_AppendTailList(&op_list, &channel_internal_ptr->op_list);
+        (void)DList_RemoveEntryList(&channel_internal_ptr->op_list);
 
```

Comment:
For my understanding, what does the RemoveEntryList do here?

Reply:
The aim here is to make a local copy of the list in `channel_internal_ptr->op_list` and set `channel_internal_ptr->op_list` to be empty.

There is no direct API that achieves this. My solution is to append the list to the local `op_list`, (this includes the head of `channel_internal_ptr->op_list`) and then remove the head node so the local `op_list` becomes the head.

Reply:
state: (op_list = [], channel_internal_ptr->op_list = [...ops..])
`DList_AppendTailList(&op_list, &channel_internal_ptr->op_list);`
state: (op_list = [channel_internal_ptr-op_list, ...ops...], channel_internal_ptr->op_list = [...ops..])
```
(void)DList_RemoveEntryList(&channel_internal_ptr->op_list);
DList_InitializeListHead(&channel_internal_ptr->op_list);
```

state: (op_list = [...ops...], channel_internal_ptr->op_list = [])

Reply:
Removing this in next iteration and copying the list entry-by-entry instead